src/main/java/pl/psnc/dei/ui/components/batches/BatchImportComponent.java

```diff
@@ -0,0 +1,211 @@
+    }
+
+    private TextArea prepareRecordsTextArea() {
+        recordsTextArea = new TextArea();
+        recordsTextArea.setMinHeight("150px");
+        recordsTextArea.setWidthFull();
+        recordsTextArea.addClassName("query-form");
+        return recordsTextArea;
+    }
+
+    private void sendImport() {
+        if (!areInputsValid()) {
+            return;
+        }
+        Set<String> retrievedRecords;
+        try {
+            retrievedRecords = retrieveRecords();
+        } catch (ParseRecordsException ex) {
+            showNotification(ex.getMessage());
+            return;
+        }
+        try {
+            String datasetId = dataset != null ? dataset.getDatasetId() : null;
+            Set<Record> uploadedRecords = batchService.uploadRecords(project.getName(), datasetId, retrievedRecords);
+            if (uploadedRecords.isEmpty()) {
+                showNotification("There are no records to upload!");
+                return;
+            }
+            Import anImport = importPackageService.createImport(importNameTextField.getValue(), project.getProjectId(), uploadedRecords);
+            showNotification("Successfully created import: " + anImport.getName());
+        } catch (NotFoundException e) {
+            showNotification("Invalid input. " + e.getMessage());
+        }
+    }
+
+    private Set<String> retrieveRecords() throws ParseRecordsException {
+        String rawInput = recordsTextArea.getValue();
+        ObjectMapper mapper = new ObjectMapper();
+        Set<String> parsedRecords;
+        try {
+            parsedRecords = mapper.readValue(rawInput, new TypeReference<Set<String>>() {
+            });
+        } catch (JsonParseException | JsonMappingException jsonProcessingException) {
+            parsedRecords = readLines(rawInput);
+        } catch (Exception e) {
+            logger.error("Unable to retrieve input records. " + e.getMessage());
+            throw new ParseRecordsException("Cannot read records!");
+        }
+        Set<String> normalizedRecords = normalizeRecords(parsedRecords);
+        normalizedRecords.forEach(record -> {
+            if (!isValidRecord(record)) {
+                throw new ParseRecordsException("Invalid record: " + record);
+            }
+        });
+        return normalizedRecords;
+    }
+
+    private Set<String> readLines(String input) {
+        String[] split = input.split("\n");
+        return Set.of(split);
+    }
+
+    private Set<String> normalizeRecords(Set<String> records) {
+        return records.stream()
+                .map(String::trim)
+                .collect(Collectors.toSet());
+    }
+
+    private boolean isValidRecord(String s) {
+        return EuropeanaRecordIdValidator.validate(s);
+    }
+
+    private boolean areInputsValid() {
+        String records = recordsTextArea.getValue();
+        if (records == null || records.isBlank()) {
+            showNotification("No records specified!");
```

Comment:
I would add more descriptive message informing about the structure of the input.

Reply:
Added description under textarea, which informs about available input types.